src/t8_cmesh/t8_cmesh_readmshfile.c

```diff
@@ -198,36 +198,53 @@ t8_cmesh_check_version_of_msh_file (FILE * fp)
     if (retval != 1) {
       t8_global_errorf
         ("Reading the msh-file in order to check the MeshFormat-number failed.\n");
-      t8_debugf ("The line is %s", line);
+      goto die_format;
     }
   }
 
   /* Got to the next line containing the MeshFormat. */
   (void) t8_cmesh_msh_read_next_line (&line, &linen, fp);
   /* Get the MeshFormat number of the file */
-  retval = sscanf (line, "%d.%d", &version_number, &sub_version_number);
+  retval =
+    sscanf (line, "%d.%d %d", &version_number, &sub_version_number,
+            &check_format);
 
   /*Checking for read/write error. */
-  if (retval != 2) {
-    t8_global_errorf ("The MeshFormat-number was not read correctly.\n");
+  if (retval != 3) {
     t8_debugf ("Reading of the MeshFormat-number failed.\n");
+    goto die_format;
+  }
+
+  /* Checks if the file is of Binary-type. */
+  if (check_format) {
+    t8_global_errorf
+      ("Incompatible file-type. t8code works with ASCII-type msh-files of version %d.\n",
```

Comment:
Aren't the msh-files of version 1 created by gmsh structured slighlty different than those of version 2? (https://gmsh.info/doc/texinfo/gmsh.html#MSH-file-format-version-1-_0028Legacy_0029)  There seems to be no MeshFormat number and the parts containing the nodes and elements are introduced by '$NOD' and '$ELM' instead of '$Nodes' and '$Elements' like in files of version 2. Apart from that, in the element section the third entry in version 2 is the 'number-of-tags' followed by these specified tags before the 'node-number-lists' starts. These tags are skipped by the function 't8_cmesh_msh_file_read_eles()'  by reading the number of tags and leaving the following entries out, whereas, in version 1 are always 3 entries before the 'node-number-list', describing the elements, starts. But they aren't preceded by the 'number-of-tags'. Therefore, the function doesn't read the elements properly, does it? So, version 1 files can't be read or am I wrong?